### PR TITLE
Fix product's feature selection

### LINKED ISSUE #16306: Improve feature creation in product page
#### Describe the bug

> Edited by @HanaRebaiQA 

There are two cases in product page : 

**Product V1 page** : There is no possibility to delete a feature when we put `Choose a value` and `Choose a feature`. The features are duplicated after save and when we put a customized value for it.

Here is a screen-record : 8.0.x with product V1 page

https://user-images.githubusercontent.com/92912932/177517930-35644b44-8d22-43f1-8c9e-83ed0550d8cc.mp4

**Product V2 page** : An exception is thrown when we put `choose a value` and `choose a feature ` 

Here is a screen-record : 8.0.x with product V2 page

https://user-images.githubusercontent.com/92912932/177517938-e58de612-24d2-4aeb-8da9-f3ba98340fee.mp4

>  //*************************The original Description************************//

If you create a feature in a product page "Add a new feature" and give it a custom value, **but without selecting an existing feature** then it is impossible to delete it after

In addition to no being able to delete it, it seems to impact existing and fine features by applying the custom value of the created feature to all the features' values.

#### Expected behavior
When you add a feature, you have a drop-down with all features. 

If you have not yet selected a feature, the “pre-defined value” drop-down is disabled and the “customized value” field should be also disabled (currently it’s not).

capture d'écran_2129

If you select a pre-defined value, the customized value should be disabled because a feature can’t have both.

capture d'écran_2130

If you start enter a customized value, the pre-defined value drop-down should be disabled because a feature can’t have both.

capture d'écran_2131

#### Steps to Reproduce

1. Go to 'Catalog > products > select any product'
2. In the product page, on the first tab, click on 'Add a feature'
3. For "Feature" and "Pre-defined value", select "Choose a feature" and add a custom value
4. Save
5. go to another page and come back to the product page
6. See the feature you created, delete it and save
7. Go to another page and come back, the feature still exist.

**Screenshots**

See the video:
https://drive.google.com/file/d/1nZPWLhODwa9FQlvmK2aHRDWNPVEy8O1W/view

#### Additional informations
PrestaShop version: 1.7.7.0
PHP version: 7.2

### PR DESCRIPTION
**Pull request by alfonsolr09**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      |  We make the boxes are deactivated if we have not selected the characteristic. If we add a custom value, the value box is disabled. If we select a value in the list, the custom value box is deactivated and deletes the value (if there was something written)
| Type?             | bug fix 
| Category?         |  BO 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #16306
| How to test?      | See features in product page
| Possible impacts? | No

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26782)

### CODE REVIEW

**Comment by alfonsolr09** [comment]
can we merge this into 1.7.8.x?

**Comment by Hlavtox** [comment]
@alfonsolr09 It won't be accepted into 1.7.8, only regressions are in the patches.

**Comment by matks** [comment]
I confirm @Hlavtox 

In the past we used to put a lot of bug fixes in patch version, including bugs "not a regression".

The consequences were
- big patches very long to validate for QA
- new issues (because the more code you ship the more bugs you add ! :trollface: )

So we dont do this anymore to have patch releases "as safe as possible".

**Review by matks — CHANGES_REQUESTED** [review]
ESLint rejects your changes please fix them 😉

**Comment by alfonsolr09** [comment]
done! 
I think I forgot something, can you run the ESLint tests again?

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 1]
```diff
@@ -396,7 +396,12 @@ const featuresCollection = (function () {
   /** Add a feature */
   function add() {
     const newForm = collectionHolder.attr('data-prototype').replace(/__name__/g, maxCollectionChildren);
-    collectionHolder.append(newForm);
+    const $newFormJquery = $(newForm);
+  // we need to put in disabled the custom
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js]:
```suggestion
    // we need to put in disabled the custom
```

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 2]
```diff
@@ -396,7 +396,12 @@ const featuresCollection = (function () {
...
+    const $customValueSelector = $newFormJquery.find('input[type=text]');
+    const $featurevalueSelector = $newFormJquery.find('select');
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js]:
```suggestion
```

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 3]
```diff
@@ -426,17 +431,45 @@ const featuresCollection = (function () {
         return url.replace(/\/\d+(?!.*\/\d+)((?=\?.*))?/, `/${newId}`);
       }
 
+    // adding events to the preloaded features
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js]:
```suggestion
      // adding events to the preloaded features
```

**Review by Progi1984 — COMMENTED** [review]
@alfonsolr09 Thank your contribution. Could you fix all ESLint errors, please ?

**Comment by alfonsolr09** [comment]
I have already solved all the errors you indicated to me, thank you

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 4]
```diff
@@ -426,17 +430,45 @@ const featuresCollection = (function () {
...
+        {
+          $selector.val(0);
+        }
+      });
+
       /** On feature selector event change, refresh possible values list */
       $(document).on('change', '.feature-collection select.feature-selector', function (event) {
         const that = event.currentTarget;
         const $row = $($(that).parents('.row')[0]);
         const $selector = $row.find('.feature-value-selector');
```
Comment (Progi1984) [inline admin-dev/themes/default/js/bundle/product/form.js:461]:
```
```

**Comment by alfonsolr09** [comment]
check it now please @Progi1984

**Comment by atomiix** [comment]
@alfonsolr09 looks like eslint is still not happy 😅

**Comment by alfonsolr09** [comment]
Can you solve it? @atomiix

**Comment by atomiix** [comment]
@alfonsolr09 I did a mistake while rebasing your PR and I cannot push on your develop branch anymore 🤦‍♂️. I pushed the changes on my repo: https://github.com/atomiix/PrestaShop/tree/alfonsolr09-develop, can you rebase your develop branch on this one so we can re-open your PR? Thanks

**Comment by alfonsolr09** [comment]
done?? https://github.com/PrestaShop/PrestaShop/pull/27097

**Comment by alfonsolr09** [comment]
check it please, i think i didn't get it right and corrupted the code ...

**Comment by atomiix** [comment]
@alfonsolr09 You created a new PR from my own branch instead of rebasing yours from mine 😅  That being said, the new PR seems valid, I don't see any corrupted code. It would have been simpler to update the original PR (still not too late if you wanna try ;)) for reviewers, but we can keep the new one too ;)

**Comment by alfonsolr09** [comment]
> instead of rebasing yours from mine 

How can i do ?? I don't really understand GitHub ...

**Comment by matks** [comment]
@alfonsolr09 we cannot teach you git neither github, you need to learn it if you want to contribute to a project hosted on github

Here are some nice doc
https://git-scm.com/docs/user-manual
https://docs.github.com/

**Pull request closed without merging**